datadog_test.go

```diff
@@ -0,0 +1,265 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package datadog
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+var (
+	measureCount = stats.Int64("fooCount", "testing count metrics", stats.UnitBytes)
+	measureSum   = stats.Int64("fooSum", "testing sum metrics", stats.UnitBytes)
+	measureLast  = stats.Int64("fooLast", "testing LastValueData metrics", stats.UnitBytes)
+	measureDist  = stats.Int64("fooHisto", "testing histogram metrics", stats.UnitDimensionless)
+	testTags     []tag.Key
+)
+
+func newView(agg *view.Aggregation) *view.View {
+	return &view.View{
+		Name:        "fooCount",
+		Description: "fooDesc",
+		Measure:     measureCount,
+		Aggregation: agg,
+	}
+}
+
+func newCustomView(measureName string, agg *view.Aggregation, tags []tag.Key, measure *stats.Int64Measure) *view.View {
+	return &view.View{
+		Name:        measureName,
+		Description: "fooDesc",
+		Measure:     measureCount,
+		TagKeys:     tags,
+		Aggregation: agg,
+	}
+}
+
+func TestExportView(t *testing.T) {
+	reportPeriod := time.Millisecond
+	exporter := testExporter(Options{})
+
+	vd := &view.Data{
+		View: newCustomView("fooCount", view.Count(), testTags, measureCount),
+	}
+	if err := view.Register(vd.View); err != nil {
+		t.Fatalf("Register error occurred: %v\n", err)
+	}
+	defer view.Unregister(vd.View)
+	// Wait for exporter to process metrics
+	<-time.After(10 * reportPeriod)
+
+	ctx := context.Background()
+	stats.Record(ctx, measureCount.M(1))
+	<-time.After(10 * time.Millisecond)
+
+	actual := exporter.statsExporter.viewData["fooCount"].View
+	if actual != vd.View {
+		t.Errorf("Expected: %v, Got: %v\n", vd, actual)
+	}
+}
+
+func TestSanitizeString(t *testing.T) {
+	testCases := []struct {
+		input string
+		want  string
+	}{
+		{"data-234_123!doge", "data_234_123_doge"},
+		{"hello!good@morn#ing$test%", "hello_good_morn_ing_test_"},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.input, func(t *testing.T) {
+			got := sanitizeString(tc.input)
+			if got != tc.want {
+				t.Errorf("Expected: %v, Got: %v\n", tc.want, got)
+			}
+		})
+	}
+}
+
+func TestSanitizeMetricName(t *testing.T) {
+	vd1 := &view.Data{
```

Comment:
I would say to only do:
```go
v1 := newCustomView("fooGauge", view.Count(), testTags, measureCount)
v2 := newCustomView("bar-Sum", view.Sum(), testTags, measureSum)
```
We don't really need the `view.Data` structure anywhere.

The same in the other tests.